app/e-app/sand/css.html

```diff
@@ -43,7 +51,8 @@
                 content = {
                     input: '',
                     expected: undefined,
-                    expectedLength: undefined
+                    expectedLength: undefined,
+                    qdidAttr: 'qdid-' + ~~(Math.random() * 100000)
```

Comment:
<img width="311" alt="screen shot 2017-09-27 at 2 56 26 pm" src="https://user-images.githubusercontent.com/6261302/30912128-13bdfd92-a394-11e7-8c67-76c84be5d3da.png">

Reply:
`~~` – still have no idea what does this construct stands for and honestly don't want to 😄

Reply:
JavaScript is _amazing_

Reply:
It's very simple! `~` simply inverts bits BUT (what is important) converts the given number to Int32 (by simple ignoring "bits after dot"). So

     3.1 -> `0` `10000000` `10001100110011001100110`
   ~3.1 -> `1` `01111111` `01111111111111111111111` (cut decimal part (fill mantissa with `0`) AND invert)
~~3.1 -> `0` `10000000` `10000000000000000000000` (invert once again to restore the remaining after "cut" original bits)

Reply:
Or another explanation: `~x = -(x+1)`
So:
x = 4.1
~x = -(4 + 1) // decimal part is removed
~~x = -(-(4 + 1) + 1) = -(-5 + 1) = -(-4) = 4

Also, just discovered, there's [Math.trunc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/trunc)

Reply:
why should we bother about bit representation of number (especially float) in case we just want to drop decimal part... 😢

Reply:
Polyfill

```
Math.trunc = Math.trunc || function(x) {
  return x - x % 1;
};
```

This looks much more clear 😄

Reply:
Agree, there're several more readable options, but:
1. `Math.round/ceil/floor` are much slower than `~~`
2. `Math.trunc` is ES2015 feature and the code inside sand is ES5 compatible as it was not really decided in which env it should be run (now we have electron and headless chrome with nice features support but there're other "backends" coming...)

Reply:
Not to offend @kirilknysh or anyone, but performance should come after correctness and readability. In this case though the code will be most certainly read by Kiril exclusively (and zandbak is not [yet] a huge OSS project with 1000s of contributors) so I guess it's okay to have `~~`.

Reply:
Sure thing, I've raised this discussion not to get rid of this line, but to know your opinion and get more insight :) And yes, I've learned smth new (wasn't aware of `Math.trunc` 🐘 )

Reply:
🐘 😸